src/auth/index.ts

```diff
@@ -298,6 +306,20 @@ export class Auth {
     );
   }
 
+  /**
+   * Upgrade hash in the database
+   * @param newHash
+   */
+  public upgradeHash(newHash: string): Promise<void> {
+    return this.client.post(
+      '/users/upgrade-hash',
```

Comment:
Why not a `PUT /users/password` (as you update the password)?

Reply:
Would the user be notified about the password change? Cause ideally he shouldn't notice the switch.  I was not sure if password change will show some message/prompt/etc., so I created a new function. Can change it to pwd change one.

Reply:
Whether the customer notices the switch or not is irrelevant to respecting REST, so we should aim to respect that. If that endpoint already exists, we may need to adapt it, but creating a new one that is a PATCH* is the best way. (`PATCH /users/:id`)

* PUT is for replacing the entire entity; we are just adding a partial modification. My bad there.

Reply:
@sg-gs  Done